experimental/batchmap/sumdb/build/map.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// mapdemo is a simple example that shows how a verifiable map can be
+// constructed in Beam.
+package main
+
+import (
+	"context"
+	"crypto"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/databaseio"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/filesystem/local"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+
+	"github.com/google/trillian/experimental/batchmap"
+	"github.com/google/trillian/experimental/batchmap/tilepb"
+	"github.com/google/trillian/merkle/coniks"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const hash = crypto.SHA512_256
```

Comment:
Could this be:
```golang
var hash = sha512.Sum512_256
```

and below just use hash(...) rather than instantiating a new Hasher?

(same in `verify.go`)

Reply:
We still need this `crypto.Hash` to pass into `batchmap.Create`. There's also the problem that in Beam having variables around requires them to be serializable to be run inside a ParDo function. Constants are easier.

What I may do in the future is come back to this and optimize it for performance, but right now it's designed to be as clear as possible and have as few pointy ends to poke your eyes out around the Beam machinery as possible.